### PR TITLE
add azurename annotation

### PR DESCRIPTION
**Pull request by iwangjintian**
For nested resource sync back, we still need an azureName annotation to make it work

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go [hunk 1]
```diff
@@ -219,6 +220,9 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Invalid solution name: catalog - %s", name), v1alpha2.BadRequest)
 					}
 
+					if label_key != "" && label_value != "" && annotation_name != "" {
```
Comment (LeiQL) [inline api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go:223]: Why adding annotation depends on label env variables?

Comment (iwangjintian) [reply api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go:223]: labeling and annotation should be bundled

Comment (msftcoderdjw) [reply api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go:223]: From OSS perspective, we can treat them separately?

**Review by Haishi2016 — COMMENTED** [review]
I don't know what's exactly needed by Azure data sync so I'll leave this to others.

File: packages/helm/symphony/templates/env-configmap.yaml [hunk 2]
```diff
@@ -9,7 +9,8 @@ data:
   API_SERVING_CA: {{ include "symphony.apiServingCA" . }}
   SYMPHONY_API_URL: {{ include "symphony.httpsUrl" . }}
   USE_SERVICE_ACCOUNT_TOKENS: "true"
-{{- if and .Values.api.labelKey .Values.api.labelValue }}
+{{- if and .Values.api.labelKey .Values.api.labelValue .Values.api.annotationKey }}
```
Comment (msftcoderdjw) [inline packages/helm/symphony/templates/env-configmap.yaml:12]: We expect we can separately provide label key/values or name annotations.

File: api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go [hunk 3]
```diff
@@ -219,6 +220,12 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Invalid solution name: catalog - %s", name), v1alpha2.BadRequest)
 					}
 
+					if annotation_name != "" {
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go:223]: You can add a function in objectMetadata to add the annotation, avoid writing the same code here.

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**